.config/eslint.mjs

```diff
@@ -0,0 +1,573 @@
+// -----------------------------------------------------------------------------
```

Comment:
this is quite a large file - it seems like it'd be much easier to maintain both with the prettier config inlined, rather than in a separate prettier config file, but also by using one of the standard presets available rather than defining every single rule.

(my personal preference is also that config is static, ie, not in a .js file, but i suspect not everyone shares this view)

Reply:
> this is quite a large file - it seems like it'd be much easier to maintain both with the prettier config inlined, rather than in a separate prettier config file

I don't want to mix them in case there are some editor plugins that look for:
- `.prettierrc`
- `.prettierrc.js`
- `.prettierrc.json`.

> but also by using one of the standard presets available rather than defining every single rule.

Are there any presets I should look into? I haven't really been shopping around.

> (my personal preference is also that config is static, ie, not in a .js file, but i suspect not everyone shares this view)

Line #388 of this file basically made the decision for me, but I also make good use of it here. I really like it myself.

Reply:
The intention is to actively prevent editors that intend to invoke prettier directly; prettier should only be ran on JS via eslint.

Reply:
> prettier should only be ran on JS via eslint

Yup, I was thinking more about for editor plugins that use it for formatting Markdown (and other) files.

Reply:
Alright, I'm willing to combine the two files since I don't really like using Prettier for formatting Markdown.

Yulia mentioned that I may want to consider taking a look at what @mozilla uses.

https://dxr.mozilla.org/mozilla-central/source/tools/lint/eslint/.eslintrc.js

They've also got a couple plugins that may prove useful.